jerry-core/ecma/operations/ecma-function-object.c

```diff
@@ -1048,6 +1048,16 @@ ecma_op_function_construct (ecma_object_t *func_obj_p, /**< Function object */
                                               arguments_list_len);
       break;
     }
+    case ECMA_OBJECT_TYPE_GENERAL:
+    {
+      /* Catch the special case when a the class extends value in null
+         and the class has no explicit constructor to raise TypeError.*/
+      JERRY_ASSERT (!ecma_op_function_has_construct_flag (arguments_list_p));
+      JERRY_ASSERT (ecma_get_object_prototype (func_obj_p) == NULL);
+
+      ret_value = ecma_raise_type_error (ECMA_ERR_MSG ("Super constructor null is not a constructor."));
+      break;
+    }
```

Comment:
Can it be moved to the `if (ext_func_obj_p->u.external_handler_cb == ecma_op_function_implicit_constructor_handler_cb)` below?

Reply:
I'd rather not move it. The `ecma_op_function_implicit_constructor_handler_cb` check requires that the object's type must be `ECMA_OBJECT_TYPE_EXTERNAL_FUNCTION`, but for the special behavior for `class extends null {}` the class extends values's type is always `ECMA_OBJECT_TYPE_GENERAL`. See the related part [here](https://github.com/pando-project/jerryscript/blob/master/jerry-core/vm/vm.c#L1331).